src/msg/async/AsyncConnection.cc

```diff
@@ -394,13 +394,8 @@ ssize_t AsyncConnection::_try_send(bool send, bool more)
 
   // trim already sent for outcoming_bl
   if (sent_bytes) {
-    if (sent_bytes < outcoming_bl.length()) {
-      bufferlist bl;
-      outcoming_bl.splice(sent_bytes, outcoming_bl.length()-sent_bytes, &bl);
-      bl.swap(outcoming_bl);
-    } else {
-      outcoming_bl.clear();
-    }
+    assert(sent_bytes <= outcoming_bl.length());
+    outcoming_bl.splice(0, sent_bytes);
```

Comment:
Yes, we want to trim already sent part, from my point of view, It seems to be easier to understand, and achieve the same effect as above (both sent_bytes < outcoming_bl.length() and sent_bytes = outcoming_bl.length()). It seems to be unnecessary to define bl and do swap operation. tks.